crud/src/main/java/com/redhat/lightblue/assoc/ep/ExecutionPlan.java

```diff
@@ -0,0 +1,474 @@
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.assoc.ep;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Collection;
+import java.util.Map;
+import java.util.HashMap;
+
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import com.redhat.lightblue.query.QueryExpression;
+import com.redhat.lightblue.query.Projection;
+import com.redhat.lightblue.query.FieldProjection;
+import com.redhat.lightblue.query.ProjectionList;
+import com.redhat.lightblue.query.Sort;
+import com.redhat.lightblue.query.NaryLogicalExpression;
+import com.redhat.lightblue.query.NaryLogicalOperator;
+
+import com.redhat.lightblue.metadata.CompositeMetadata;
+import com.redhat.lightblue.metadata.ResolvedReferenceField;
+import com.redhat.lightblue.metadata.ObjectField;
+import com.redhat.lightblue.metadata.ArrayField;
+import com.redhat.lightblue.metadata.ArrayElement;
+import com.redhat.lightblue.metadata.ObjectArrayElement;
+import com.redhat.lightblue.metadata.ReferenceField;
+import com.redhat.lightblue.metadata.FieldTreeNode;
+import com.redhat.lightblue.metadata.FieldCursor;
+
+import com.redhat.lightblue.assoc.QueryPlan;
+import com.redhat.lightblue.assoc.QueryPlanNode;
+import com.redhat.lightblue.assoc.Conjunct;
+import com.redhat.lightblue.assoc.RewriteQuery;
+import com.redhat.lightblue.assoc.QueryPlanData;
+import com.redhat.lightblue.assoc.QueryFieldInfo;
+
+import com.redhat.lightblue.eval.SortFieldInfo;
+
+import com.redhat.lightblue.util.JsonUtils;
+import com.redhat.lightblue.util.Path;
+
+/**
+ * Execution plan is a tree of execution blocks. Every node in the
+ * query plan is converted to an execution block, and connected in the
+ * same way query plan nodes are connected. Each execution block
+ * contains a pipeline, and every step of the pipeline performs a step
+ * of the operation.
+ *
+ * The search plan and retrieval plan are constructed differently.
+ *
+ * In the retrieval plan, the root entity is at the root, and every
+ * destination node is a child node. So, the assembler first runs the
+ * parent node, gets the docs, and then runs the child nodes and
+ * attaches the documents to the slots.
+ *
+ * Query plan is optional, and it has a different layout than the
+ * retrieval plan. All that we do with the query plan is to retrieve
+ * the root node documents, so query plan does not retrieve all
+ * associated entities.
+ */
+public class ExecutionPlan {
+
+    static private final Logger LOGGER=LoggerFactory.getLogger(ExecutionPlan.class);
+    
+    private final CompositeMetadata rootMd;
```

Comment:
It is used. It is passed to all execution blocks.

On Wed, Mar 30, 2016 at 12:50 PM, Brandon Vulaj <notifications@github.com>
wrote:

> In crud/src/main/java/com/redhat/lightblue/assoc/ep/ExecutionPlan.java
> <https://github.com/lightblue-platform/lightblue-core/pull/608#discussion_r57942605>
> :
>
> > + *
> > + * In the retrieval plan, the root entity is at the root, and every
> > + * destination node is a child node. So, the assembler first runs the
> > + * parent node, gets the docs, and then runs the child nodes and
> > + * attaches the documents to the slots.
> > + *
> > + * Query plan is optional, and it has a different layout than the
> > + * retrieval plan. All that we do with the query plan is to retrieve
> > + * the root node documents, so query plan does not retrieve all
> > + * associated entities.
> > + */
> > +public class ExecutionPlan {
> > +
> > +    static private final Logger LOGGER=LoggerFactory.getLogger(ExecutionPlan.class);
> > +
> > +    private final CompositeMetadata rootMd;
>
> This is never actually used. Gets set to param then the ctor just uses the
> param.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/608/files/4ed1ac30c567cdd5b0c9b29591a5eac893d9a412#r57942605>
>